src/SUnitRange.jl

```diff
@@ -19,7 +19,7 @@ end
 @pure SUnitRange(a::Int, b::Int) = SUnitRange{a, max(0, b - a + 1)}()
 
 @pure @propagate_inbounds function getindex(x::SUnitRange{Start, L}, i::Int) where {Start, L}
-    @boundscheck if i < Start || i >= (Start + L)
+    @boundscheck if i < 1 || i > L
```

Comment:
Now with extra testing :-)  I only used this because I wanted the general recursive version to work... but I got hung up on numerical stability issues.